core/src/main/java/com/linecorp/armeria/client/proxy/ProxyConfig.java

```diff
@@ -201,6 +191,12 @@ public static ProxyConfig direct() {
     @Nullable
     public abstract InetSocketAddress proxyAddress();
 
+    /**
+     * Returns a new proxy address instance that respects DNS TTL.
+     * @param newProxyAddress the inet socket address
+     */
+    public abstract ProxyConfig withNewProxyAddress(InetSocketAddress newProxyAddress);
```

Comment:
Conventionally, we don't add `new` keyword for `withXXX` method.
https://github.com/line/armeria/blob/206872ced21d96eafdb1f3140d704f22731e9870/core/src/main/java/com/linecorp/armeria/client/Endpoint.java#L623-L623
```suggestion
    public abstract ProxyConfig withProxyAddress(InetSocketAddress newProxyAddress);
```